### DISCUSSION TITLE
Stepper: dynamic length not supported

### DISCUSSION THREAD

**Discussion by Etchelon** [skeletonlabs/skeleton#835] [Ideas]
I noticed that the Stepper component doesn't respond to a change in the length prop.
I have a complex form where some steps are visible based on what you select in previous steps, and the Stepper just breaks when some steps are added.
I am currently able to work around this by wrapping the whole stepper inside a `{#key}` block:
```
{#key steps}
    <Stepper active={activeStep$} length={steps.length} on:complete={save}>
        [...]
    </Stepper>
{/key}
```
where steps is an array of strings with the name of the steps to show.
I glimpsed at the source code and it's probably enough to turn the `length` context prop into a writable, which the stepper can update when its input prop changes.

**Comment by endigo9740** [comment]
We have a proposed rewrite to adjust a lot of the UI/UX of this component in the near future:

- https://github.com/skeletonlabs/skeleton/issues/784

This would be useful to know during that process, so I'll link to this issue as reference. Thanks!
